api/src/main/scala/hmda/api/http/InstitutionsHttpApi.scala

```diff
@@ -0,0 +1,200 @@
+
+  implicit val timeout: Timeout
+
+  val splitLines = Framing.delimiter(ByteString("\n"), 2048, allowTruncation = true)
+
+  val institutionsPath =
+    path("institutions") {
+      val institutionsActor = system.actorSelection("/user/institutions")
+      get {
+        val fInstitutions = (institutionsActor ? GetState).mapTo[Set[Institution]]
+        onComplete(fInstitutions) {
+          case Success(institutions) =>
+            complete(ToResponseMarshallable(Institutions(institutions)))
+          case Failure(error) =>
+            log.error(error.getLocalizedMessage)
+            complete(HttpResponse(StatusCodes.InternalServerError))
+        }
+      }
+    }
+
+  val institutionByIdPath =
+    path("institutions" / Segment) { fid =>
+      extractExecutionContext { executor =>
+        val institutionsActor = system.actorSelection("/user/institutions")
+        val filingsActor = system.actorOf(FilingPersistence.props(fid))
+        get {
+          implicit val ec: ExecutionContext = executor
+          val fInstitutionDetails = institutionDetails(fid, institutionsActor, filingsActor)
+          onComplete(fInstitutionDetails) {
+            case Success(institution) =>
+              filingsActor ! Shutdown
+              complete(ToResponseMarshallable(institution))
+            case Failure(error) =>
+              filingsActor ! Shutdown
+              log.error(error.getLocalizedMessage)
+              complete(HttpResponse(StatusCodes.InternalServerError))
+          }
+        }
+      }
+    }
+
+  val filingByPeriodPath =
+    path("institutions" / Segment / "filings" / Segment) { (fid, period) =>
+      extractExecutionContext { executor =>
+        val filingsActor = system.actorOf(FilingPersistence.props(fid))
+        val submissionActor = system.actorOf(SubmissionPersistence.props(fid, period))
+        get {
+          implicit val ec: ExecutionContext = executor
+          val fDetails: Future[FilingDetail] = filingDetailsByPeriod(period, filingsActor, submissionActor)
+          onComplete(fDetails) {
+            case Success(filingDetails) =>
+              filingsActor ! Shutdown
+              submissionActor ! Shutdown
+              complete(ToResponseMarshallable(filingDetails))
+            case Failure(error) =>
+              filingsActor ! Shutdown
+              submissionActor ! Shutdown
+              complete(HttpResponse(StatusCodes.InternalServerError))
+          }
+        }
+      }
+    }
+
+  val submissionPath =
+    path("institutions" / Segment / "filings" / Segment / "submissions") { (fid, period) =>
+      val submissionsActor = system.actorOf(SubmissionPersistence.props(fid, period))
+      post {
+        val submissionsActor = system.actorOf(SubmissionPersistence.props(fid, period))
+        submissionsActor ! CreateSubmission
+        val fLatest = (submissionsActor ? GetLatestSubmission).mapTo[Submission]
+        onComplete(fLatest) {
+          case Success(submission) =>
```

Comment:
We might need to do something like check the id of the submission returned by `GetLatestSubmission` to make sure that it is as expected which might require something like assigning the id of the submission from `CreateSubmission` as the date/time or if we want to keep using the incrementing method we might want to query want the current is and checking that the new is 1 above it. If we knew the id in the `submissionPath` it might be cleaner to query for the submission of that id rather than looking at the latest one.